apps/chat-e2e/src/tests/ChatSwitch.test.ts

```diff
@@ -0,0 +1,160 @@
+    conversationData,
+    dataInjector,
+    setTestIds,
+    localStorageManager,
+    chat,
+    chatMessages,
+    compareConversationSelector,
+    compareConversation,
+  }) => {
+    setTestIds(
+      'EPMRTC-598',
+      'EPMRTC-599',
+      'EPMRTC-600',
+      'EPMRTC-601',
+      'EPMRTC-602',
+    );
+    const request =
+      'give me a sci-fi story with a main topic of your choice. 200 tokens minimum';
+    let firstConversation: Conversation;
+    let secondConversation: Conversation;
+    let thirdConversationName;
+
+    const verifyConversationCursor = async (
+      conversationName: string,
+      expectedCursor: string,
+    ) => {
+      await conversations.getEntityByName(conversationName).hover();
+      const style = await conversations
+        .getConversationName(conversationName)
+        .getComputedStyleProperty(Styles.cursor);
+      expect
+        .soft(style[0], `Conversation cursor is ${expectedCursor}`)
+        .toBe(expectedCursor);
+    };
+
+    const verifySelectedConversation = async (conversationName: string) => {
+      await expect
+        .soft(
+          conversations.selectedConversation(conversationName),
+          ExpectedMessages.conversationIsSelected,
+        )
+        .toBeVisible();
+    };
+
+    await dialTest.step('Prepare two empty conversations', async () => {
+      firstConversation = conversationData.prepareEmptyConversation();
+      conversationData.resetData();
+      secondConversation = conversationData.prepareEmptyConversation();
+      await dataInjector.createConversations([
+        firstConversation,
+        secondConversation,
+      ]);
+      await localStorageManager.setSelectedConversation(firstConversation);
+    });
+
+    await dialTest.step('Send request to the first conversation', async () => {
+      await dialHomePage.openHomePage();
+      await dialHomePage.waitForPageLoaded();
+      await dialHomePage.throttleAPIResponse(
+        API.chatHost,
+        responseThrottlingTimeout,
+      );
+      await chat.sendRequestWithButton(request, false);
+      firstConversation.name = request;
+    });
+
+    await dialTest.step(
+      'Verify any conversation cursor is "not-allowed" during the text generation',
+      async () => {
+        await verifyConversationCursor(
+          firstConversation.name,
+          Cursors.notAllowed,
+        );
+        await verifyConversationCursor(
+          secondConversation.name,
+          Cursors.notAllowed,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Verify another conversation is not selectable during text generation',
+      async () => {
+        await conversations.getEntityByName(secondConversation.name).click();
+        await verifySelectedConversation(firstConversation.name);
+      },
+    );
+
+    await dialTest.step(
+      'Select "Replay" option from the first conversation menu',
+      async () => {
+        await chatMessages.waitForResponseReceived();
```

Comment:
the response takes time, to avoid waiting, I see 2 solutios:
- click 'Stop generating' btn immediately
- mock API response using method:
`await dialHomePage.mockChatTextResponse(MockedChatApiResponseBodies.simpleTextBody);`

Reply:
done